contracts/lib/SpongePoseidon.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+import "../lib/Poseidon.sol";
+
+contract SpongePoseidon {
+    uint32 constant BATCH_SIZE = 5;
+    uint32 constant HASH_FN_BATCH_SIZE = 6;
+
+    function getValueByIndex(
+        uint256[] calldata arr,
+        uint32 idx,
+        uint32 length
+    ) internal pure returns (uint256) {
+        if (idx < length) {
+            return arr[idx];
+        }
+        return 0;
+    }
+
+    function hash(uint256[] calldata values) public view returns (uint256) {
+        uint32 iterationCount = 0;
+        uint32 length = uint32(values.length);
+        uint256 fullHash = PoseidonUnit6L.poseidon(
+            [
+                getValueByIndex(values, 0, length),
+                getValueByIndex(values, 1, length),
+                getValueByIndex(values, 2, length),
+                getValueByIndex(values, 3, length),
+                getValueByIndex(values, 4, length),
+                getValueByIndex(values, 5, length)
+            ]
+        );
+
+        uint32 restLength = length - HASH_FN_BATCH_SIZE;
+        if (restLength > BATCH_SIZE) {
+            uint32 r = restLength % BATCH_SIZE;
+            uint32 diff = BATCH_SIZE - r;
+            iterationCount = (restLength + diff) / BATCH_SIZE;
+        }
```

Comment:
@Kolezhniuk, I guess you can try OpenZeppelin [ceilDiv(a, b)](https://docs.openzeppelin.com/contracts/4.x/api/utils#Math-ceilDiv-uint256-uint256-) method instead of this 5 lines o code

Reply:
@AndriianChestnykh not sure how this function helps here
I guess here is fine to have `iterationCount` as float as for me